testsuite/MDAnalysisTests/coordinates/test_netcdf.py

```diff
@@ -181,20 +182,36 @@ def tearDown(self):
         del self.Writer
         del self.tmpdir
 
-    def test_write_trajectory(self):
+    def _test_write_trajectory(self):
         t = self.universe.trajectory
         with self.Writer(self.outfile, t.n_atoms, dt=t.dt) as W:
             self._copy_traj(W)
         self._check_new_traj()
-        import netCDF4
         #for issue #518 -- preserve float32 data in ncdf output
-        dataset = netCDF4.Dataset(self.outfile, 'r', format='NETCDF3')
+        # NOTE: This originally failed with the dtype('>f4') instead
+        #       of dtype('<f4') == dtype('f') == np.float32, i.e. then
+        #       endianness is different. The current hack-ish solution
+        #       ignores endianness by comparing the name of the types,
+        #       which should be "float32".
+        #       See http://docs.scipy.org/doc/numpy-1.10.0/reference/arrays.dtypes.html
+        #       and https://github.com/MDAnalysis/mdanalysis/pull/503
+        dataset = netcdf.netcdf_file(self.outfile, 'r')
         coords = dataset.variables['coordinates']
         time = dataset.variables['time']
-        assert_equal(coords.dtype, np.float32,
-                     err_msg='ncdf coord output not float32')
-        assert_equal(time.dtype, np.float32,
-                     err_msg='ncdf time output not float32')
+        assert_equal(coords[:].dtype.name, np.dtype(np.float32).name,
+                     err_msg='ncdf coord output not float32 '
+                             'but {}'.format(coords[:].dtype))
+        assert_equal(time[:].dtype.name, np.dtype(np.float32).name,
+                err_msg='ncdf time output not float32 '
+                        'but {}'.format(time[:].dtype))
+
+    def test_write_trajectory_netCDF4(self):
+        pytest.importorskip("netCDF4")
+        return self._test_write_trajectory()
+
+    @block_import('netCDF4')
```

Comment:
Looking at coveralls it becomes clear that testing WITHOUT netcdf4 is not working. The code path where we fall back to netcdf is not tested.

Probably I don't understand how I should be using `@block_import` – presumably it will not remove existing imports but only act when `import ...`. The way this test is written, the `netCDF4` module has already been loaded.